db/collab.go

```diff
@@ -36,25 +37,36 @@ func init() {
 
 func (s *Session) AddConn(uid string, conn *websocket.Conn) error {
 	s.Lock.Lock()
+	if s.Conns[uid] != nil {
+		s.Lock.Unlock()
+		return errors.New("User is already connected")
+	}
 	s.Conns[uid] = conn
 	s.Lock.Unlock()
 	return nil
 }
 
 func (s *Session) RemoveConn(uid string) error {
 	s.Lock.Lock()
+	if s.Conns[uid] != nil {
+		s.Lock.Unlock()
+		return errors.New("Could not remove unconnected user")
+	}
 	delete(s.Conns, uid)
 	s.Lock.Unlock()
 	return nil
 }
 
 func (s *Session) Broadcast(msg Message) error {
+	var mostRecentError error = nil
```

Comment:
```suggestion
	var mostRecentError error
```

The zero value of the error type is, itself, nil.